core/common/src/ConcurrentQuery.ts

```diff
@@ -0,0 +1,412 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { BentleyError, CompressedId64Set, DbResult, Id64Set, Id64String } from "@itwin/core-bentley";
+import { Point2d, Point3d } from "@itwin/core-geometry";
+
+/**
+ * Specifies the format of the rows returned by the `query` and `restartQuery` methods of
+ * [IModelConnection]($frontend), [IModelDb]($backend), and [ECDb]($backend).
+ *
+ * @public
+ * */
+export enum QueryRowFormat {
+  /** Each row is an object in which each non-null column value can be accessed by its name as defined in the EC schema.
+   * Null values are omitted.
+   */
+  UseECSqlPropertyNames,
+  /** Each row is an object in which each non-null column value can be accessed by a [remapped property name]($docs/learning/ECSqlRowFormat.md).
+   * This format is backwards-compatible with the format produced by iModel.js version 2.x. Null values are omitted.
+   */
+  UseJsPropertyNames,
+  /** Each row is an array of values accessed by an index corresponding to the property's position in the ECSql SELECT statement.
+   * Null values are included if they are followed by a non-null column, but trailing null values at the end of the array are omitted.
+   */
+  UseArrayIndexes,
+  /** The default format if none is specified by the caller. */
+  Default = UseArrayIndexes,
```

Comment:
`Default` is unnecessary - the default should be specified as default argument in function signatures.